.azure-pipelines/build-job.yml

```diff
@@ -0,0 +1,203 @@
+parameters:
+
+- name: jobName
+  type: string
+
+- name: displayName
+  type: string
+
+- name: pool
+  type: object
+
+- name: container
+  type: string
+  default: ''
+
+- name: timeoutInMinutes
+  type: number
+  default: 0
+
+- name: branch
+  type: string
+  default: ''
+
+- name: os
+  type: string
+  values:
+  - win
+  - linux
+  - osx
+  - rhel.6
+
+- name: arch
+  type: string
+  values:
+  - arm
+  - arm64
+  - x64
+  - x86
+
+- name: functionalTests
+  type: boolean
+  default: true
+
+- name: codeCoverage
+  type: boolean
+  default: false
+
+- name: componentDetection
+  type: boolean
+
+- name: sign
+  type: boolean
+
+- name: verifySigning
+  type: boolean
+  default: false
+
+- name: publishArtifact
+  type: boolean
+
+- name: enableADOLogIssue
+  type: boolean
+  default: false
+
+
+jobs:
+
+- job: ${{ parameters.jobName }}
+  displayName: ${{ parameters.displayName }}
+  pool: ${{ parameters.pool }}
+
+  ${{ if ne(parameters.container, '') }}:
+    container: ${{ parameters.container }}
+
+  ${{ if gt(parameters.timeoutInMinutes, 0) }}:
+    timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+
+  variables:
+    ${{ if eq(parameters.os, 'win') }}:
+      devCommand: dev.cmd
+    ${{ if ne(parameters.os, 'win') }}:
+      devCommand: ./dev.sh
+    layoutRoot: '_layout/${{ parameters.os }}-${{ parameters.arch }}'
+
+  steps:
+
+  # Component detection
+  - ${{ if eq(parameters.componentDetection, true) }}:
+    - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+      displayName: 'Component Detection'
+
+  # Clean
+  - checkout: self
+    clean: true
+
+  # Switch to branch if set
+  - ${{ if ne(parameters.branch, '') }}:
+    - template: switch-branch.yml
+      parameters:
+        branch: ${{ parameters.branch }}
+
+  # Build agent layout
+  - script: ${{ variables.devCommand }} layout Release
+    workingDirectory: src
+    displayName: Build & Layout Release
+    ${{ if eq(parameters.enableADOLogIssue, true) }}:
+      env:
+        ADO_ENABLE_LOGISSUE: true
```

Comment:
I don't think this needs to be wrapped in an if, we should just be able to say:
```
env:
   ADO_ENABLE_LOGISSUE: ${{ parameters.enableADOLogIssue }}

Reply:
Unfortunately, no. The toString value of the underlying boolean value will be "True" and the logic around ADO_ENABLE_LOGISSUE is case sensitive: https://github.com/microsoft/azure-pipelines-agent/blob/master/src/dev.sh#L74